ext/node/polyfills/http2.ts

```diff
@@ -1488,24 +1529,53 @@ export function connect(
     host = authority.host;
   }
 
+  let conn, nodeConn, url;
+
+  console.log("authority", authority);
   // TODO(bartlomieju): handle defaults
   if (typeof options.createConnection === "function") {
-    console.error("Not implemented: http2.connect.options.createConnection");
+    // console.error("Not implemented: http2.connect.options.createConnection");
     // notImplemented("http2.connect.options.createConnection");
+    nodeConn = options.createConnection(host, options);
+    url = `http://${host}${port == 80 ? "" : (":" + port)}`;
+  } else {
+    // TODO(bartlomieju): using `localhost` as a `host` refuses to connect
+    // when server is listening on `0.0.0.0`
+    console.log(protocol, host, port);
+
+    if (protocol == "http:") {
+      conn = Deno.connect({ port, hostname: "127.0.0.1" });
+      url = `http://${host}${port == 80 ? "" : (":" + port)}`;
+    } else if (protocol == "https:") {
+      conn = Deno.connectTls({ port, hostname: host, alpnProtocols: ["h2"] });
+      url = `https://${host}${port == 443 ? "" : (":" + port)}`;
+    } else {
+      throw new TypeError("Unexpected URL protocol");
+    }
```

Comment:
This bit should be rewritten to always use Node APIs for creating sockets so we can unify conditional below